src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Components/Identity/LogoutForm.razor

```diff
@@ -0,0 +1,34 @@
+﻿@using Microsoft.AspNetCore.Identity
+@using BlazorWeb_CSharp.Data
+@using BlazorWeb_CSharp.Identity
+
+@inject SignInManager<ApplicationUser> SignInManager
+@inject NavigationManager NavigationManager
+@inject IdentityRedirectManager RedirectManager
+
+<form @onsubmit="OnSubmitAsync" @formname="logout" method="post" @attributes="AdditionalAttributes">
+    <AntiforgeryToken />
+    <input type="hidden" name="ReturnUrl" value="@NavigationManager.Uri" />
+</form>
+
+@code {
+    [Parameter(CaptureUnmatchedValues = true)]
+    public IDictionary<string, object?>? AdditionalAttributes { get; set; }
+
+    [SupplyParameterFromForm]
+    private string? ReturnUrl { get; set; }
+
+    [CascadingParameter]
+    private HttpContext HttpContext { get; set; } = default!;
```

Comment:
Do we prefer the following? I've seen people using this and think it looks better:

```suggestion
    private required HttpContext HttpContext { get; set; }
```

Reply:
We do have https://github.com/dotnet/aspnetcore/issues/47864 that tracks adding an analyzer warning if this is done. Part of the motivation is that the `required` keyword would prevent code-generated parameter setting from working, if we ever decided to do that.

Reply:
This was one of the reasons, there are more reasons, which is that this is actually "wrong" and will disable legitimate null checks that would be required for correctness.